lib/ts/recipe/passwordless/components/features/linkClickedScreen/index.tsx

```diff
@@ -114,6 +126,27 @@ const LinkClickedScreen: React.FC<PropType> = (props) => {
     const childProps = {
         recipeImplementation: props.recipe.recipeImpl,
         config: props.recipe.config,
+        showButton,
+        consumeCode: async () => {
+            const preAuthSessionId = getQueryParams("preAuthSessionId");
+            const linkCode = getURLHash();
+
+            if (preAuthSessionId === null || preAuthSessionId.length === 0 || linkCode.length === 0) {
+                // This should never happen, and even if it does the we should be already redirecting
+                throw new Error("Called consumeCode withouth link info");
+            }
+
+            try {
+                const consumeResp = await props.recipe.recipeImpl.consumeCode({
```

Comment:
Shouldn't this actually call the modified version of consumeCode you have defined above?

Reply:
No, that would return `REQUIRES_INTERACTION` again, and the other check is also handled here (without repeating the redirection code). Plus this one updates if `userContext` updates which is not the case for the other.

Reply:
In that case, I think we should rename the `consumeCode` function name on line 41. I was very confused for a while.